arangod/RocksDBEngine/RocksDBMetaCollection.cpp

```diff
@@ -614,6 +616,35 @@ Result RocksDBMetaCollection::bufferTruncate(rocksdb::SequenceNumber seq) {
   return res;
 }
 
+Result RocksDBMetaCollection::setObjectIds(std::uint64_t plannedObjectId,
+                                           std::uint64_t plannedTempObjectId) {
+  Result res;
+  auto& server = _logicalCollection.vocbase().server();
+  auto& selector = server.getFeature<EngineSelectorFeature>();
+  auto& engine = selector.engine<RocksDBEngine>();
+
+  if (plannedObjectId == _objectId && plannedTempObjectId != _tempObjectId) {
+    // just temp id has changed
+    std::uint64_t oldId = (plannedTempObjectId == 0) ? _tempObjectId : 0;
+    _tempObjectId = plannedTempObjectId;
+    if (oldId != 0) {  // need to clean up the old range
+      RocksDBKeyBounds bounds = RocksDBKeyBounds::CollectionDocuments(oldId);
+      res = rocksutils::removeLargeRange(engine.db(), bounds, true, true);
+    }
+  } else if (plannedTempObjectId != _tempObjectId) {
+    TRI_ASSERT(plannedObjectId != _objectId);
+    TRI_ASSERT(plannedObjectId != 0);
+    TRI_ASSERT(plannedObjectId = _tempObjectId);
```

Comment:
I think this in untentional, as the assert will have side-effects apart from raising an assertion failure!

Reply:
dito

Reply:
Good catch.